actiontext/lib/action_text/document.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module ActionText
+  module Document
+    extend ActiveSupport::Autoload
+    extend self
+
+    autoload :Nokogiri
+
+    attr_writer :adapter
+    delegate_missing_to :adapter
+
+    def adapter
+      @adapter ||= ActionText::Document::Nokogiri
+    end
+  end
+end
+
+ActiveSupport.run_load_hooks :action_text_document, ActionText::Document
```

Comment:
This new load hook needs to be documented in https://github.com/rails/rails/blob/main/guides/source/engines.md#available-load-hooks